### DISCUSSION TITLE
Is there a way to get the inside of capturing groups and to match multiple times?

### DISCUSSION THREAD

**Discussion by VioletDarkKitty** [rust-lang/regex#816] [Q&A] [answered]
Have I just missed it in the documentation?

```
lazy_static! {
    static ref RE: Regex = Regex::new(r"^\((.*?)\)").expect("Invalid regex");
}
```

When I use this with (a)(b)(c) and

```
for matches in RE.find_iter(...) {
   ...
}
```
I only get the first match, it's also surrounded with () still. Do you just have to keep slicing the input using the returned end value?

**Comment by BurntSushi** [comment, marked as answer]
It's difficult for me to help you here because I can't parse your question (there are multiple ways of interpreting it), and you don't say what output you want or what problem you're trying to solve.

So given that, I'll ask, have you looked at the documentation for capturing groups? It's in one of the first examples: https://docs.rs/regex/1.5.4/regex/#example-replacement-with-named-capture-groups

And there is more here: https://docs.rs/regex/1.5.4/regex/struct.Regex.html#method.captures and https://docs.rs/regex/1.5.4/regex/struct.Regex.html#method.captures_iter

**Reply by VioletDarkKitty** [reply]
I've read that part of the documentation. 

I was trying to get every string between brackets in the text. So in the above example, a and b and c from (a)(b)(c), but only the text, excluding the brackets. The capture group is around the text only but the output from the iter has brackets and only returns a single group.

**Reply by BurntSushi** [reply]
I guess I'm not following why the `captures_iter` example isn't enough here. Notably, I linked to `captures` and `captures_iter`, but... you're using `find_iter`... The example shows you exactly what you want to do as far as I can tell:

```rust
use regex::Regex;

fn main() {
    let re = Regex::new(r"\((.*?)\)").unwrap();
    let text = "(a)(b)(c)";
    for caps in re.captures_iter(text) {
        println!("inner: {:?}", &caps[1]);
    }
}
```

Output:

```
inner: "a"
inner: "b"
inner: "c"
```

Playground link: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=be5dd556b3e9040e9323acf2821107f8
